src/commands/epgsql_cmd_start_replication.erl

```diff
@@ -65,8 +65,16 @@ execute(Sock, #start_repl{slot = ReplicationSlot, callback = Callback,
 %% CopyBothResponse
 handle_message(?COPY_BOTH_RESPONSE, _Data, Sock, _State) ->
     {finish, ok, ok, epgsql_sock:set_packet_handler(on_replication, Sock)};
+handle_message(?ERROR, #error{codename = object_in_use} = Error, Sock, State) ->
+    %% A replication slot is in use already. In this case, protocol sends a ReadyForQuery message.
+    %% Adds an error to results to handle it later in the ?READY_FOR_QUERY branch.
+    Result = {error, Error},
+    {add_result, Result, Result, Sock, State};
 handle_message(?ERROR, Error, _Sock, _State) ->
     Result = {error, Error},
     {sync_required, Result};
+handle_message(?READY_FOR_QUERY, _Data, Sock, _State) ->
+    [Error = {error, _}] = epgsql_sock:get_results(Sock), % assert a single error response
+    {sync_required, Error};
```

Comment:
I wonder if it is really in a `sync required` state... What would happen if you return `{finish, Error, done, State}` and then would try to `start_replication/5` to a different, non-occupied slot? Would this 2nd attempt just work or connection is indeed out-of-sync and unusable?

Reply:
It makes sense to me to assume that the connection is in sync here.